### PR TITLE
fix(gatsby-transformer-remark): Handle headings with nested text

### LINKED ISSUE #11879: [gatsby-transformer-remark] null heading when in backticks
## Description

It's related to #5436. Opening a new issue per @KyleAMathews' [comment](https://github.com/gatsbyjs/gatsby/issues/5436#issuecomment-419614254).

When headings are in backticks (e.g. <code># \`foo\`</code>), the `headings` array from `markdownRemark` have `null` value.

### Steps to reproduce

This is the minimum possible reproduction I could come with. Hope it's enough.
https://codesandbox.io/s/502pwvro94

### Expected result

`headings` should be:
```json
[
  {
    "value": "Normal title",
    "depth": 1
  },
  {
    "value": "Code title", <--------
    "depth": 1
  }
]
```

### Actual result

```json
[
  {
    "value": "Normal title",
    "depth": 1
  },
  {
    "value": null, <--------
    "depth": 1
  }
]
```

### Environment

```
  System:
    OS: Linux 4.17 Debian GNU/Linux 9 (stretch) 9 (stretch)
    CPU: (8) x64 Intel Xeon Processor (Skylake, IBRS)
    Shell: 4.4.12 - /bin/bash
  Binaries:
    Node: 8.14.0 - /usr/local/bin/node
    Yarn: 1.12.3 - /usr/local/bin/yarn
    npm: 6.4.1 - /usr/local/bin/npm
  Languages:
    Python: 2.7.13 - /usr/bin/python
  npmPackages:
    gatsby: ^2.1.4 => 2.1.4
    gatsby-image: ^2.0.29 => 2.0.29
    gatsby-plugin-manifest: ^2.0.17 => 2.0.17
    gatsby-plugin-offline: ^2.0.23 => 2.0.23
    gatsby-plugin-react-helmet: ^3.0.6 => 3.0.6
    gatsby-plugin-sharp: ^2.0.20 => 2.0.20
    gatsby-source-filesystem: ^2.0.20 => 2.0.20
    gatsby-transformer-remark: 2.2.5 => 2.2.5
    gatsby-transformer-sharp: ^2.1.13 => 2.1.13
```

### PR DESCRIPTION
**Pull request by diegohaz**
## Description

Here's a fix. If there's a better way to write the solution, just let me know.

## Related Issues

Fixes #11879

### CODE REVIEW

File: packages/gatsby-transformer-remark/src/__tests__/extend-node.js [hunk 1]
```diff
@@ -559,3 +559,57 @@ This is [a reference]
...
+
+## \`second title\`
+`,
+    `headings {
+      value
+      depth
+    }`,
+    node => {
+      expect(node).toMatchSnapshot()
+      expect(node.headings).toEqual([
```
Comment (DSchau) [inline packages/gatsby-transformer-remark/src/__tests__/extend-node.js]: Nice idea here with this failing test 👏

**Comment by stefanprobst** [comment]
I'm sorry I missed this one before #11888 
Anyway, we should make sure headings like the following are handled correctly:
```markdown
# An **important** heading with `inline code` and text
```

**Comment by stefanprobst** [comment]
@diegohaz I'll close #11888  in favor of this. We probably want to use[ `mdast-util-to-string`](https://github.com/syntax-tree/mdast-util-to-string) here, which handles formatted text as well as inlineCode.

**Comment by diegohaz** [comment]
Thank you @stefanprobst. Added a test case with your example and fixed with `mdast-util-to-string`.

**Comment by sidharthachatterjee** [comment]
Published in gatsby-transformer-remark@2.2.6

**Pull request merged**